src/mod/endpoints/mod_sofia/sofia_presence.c

```diff
@@ -2987,12 +2987,30 @@ static int sofia_presence_sub_callback(void *pArg, int argc, char **argv, char *
 
 			if (!strcasecmp(astate, "early") || !strcasecmp(astate, "confirmed")) {
 
+				char *cmd;
+				char *callee_id_name = NULL;
+				switch_stream_handle_t user_data_stream = { 0 };
+
 				clean_to_user = switch_mprintf("%s", sub_to_user ? sub_to_user : to_user);
 				clean_from_user = switch_mprintf("%s", from_id ? from_id : from_user);
 
+				SWITCH_STANDARD_STREAM(user_data_stream);
+				cmd = switch_mprintf("%s@%s var effective_caller_id_name", clean_to_user, host);
+				if (switch_api_execute("user_data", cmd, NULL, &user_data_stream) == SWITCH_STATUS_SUCCESS) {
+					if (strncmp("-ERR", user_data_stream.data, 4)) {
+						callee_id_name = switch_mprintf("%s", (char *)user_data_stream.data);
```

Comment:
This is a memory leak.

Reply:
I changed this to switch_copy_string and pushed again.

Reply:
Why? Could you just free it later?

Reply:
It has been freed at 3083.

I thought you were concerned about a dangling pointer with the reassignment of callee_id_name, and TBH didn't understand your comment here.

Anyway switch_copy_string doesn't null terminate a truncated string so the code shouldn't be merged as it is now regardless.

Shall I revert to switch_mprintf and can you clarify what the memory leak is, or I have a lab up now with switch_snprinf and will push once I confirm it works.

Reply:
I could have github not updating the change to me properly. So please revert and I will take a look again.